ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Git/CommitsSample.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.TeamFoundation.SourceControl.WebApi;
+
+namespace Microsoft.TeamServices.Samples.Client.Git
+{
+    [ClientSample(GitWebApiConstants.AreaName, "commits")]
+    public class CommitsSample : ClientSample
+    {
+        [ClientSampleMethod]
+        public List<GitCommitRef> GetAllCommits()
+        {
+            GitRepository repo = GitSampleHelpers.FindAnyRepositoryOnAnyProject(this.Context);
+
+            return this.Context.Connection.GetClient<GitHttpClient>()
+                .GetCommitsAsync(repo.Id, new GitQueryCommitsCriteria()).Result;
```

Comment:
Ideally we should use project.Id, repo.Id to force the Urls used to be project scoped.

Reply:
I'll be changing a bunch to this as I work through them so you can leave this one for now if you want.

Reply:
Sounds good.